internal/cow/cow.go

```diff
@@ -0,0 +1,38 @@
+package cow
+
+import (
+	"io"
+
+	"github.com/Microsoft/hcsshim/internal/schema1"
+)
+
+// Process is the interface for an OS process running in a container or utility VM.
+type Process interface {
+	io.Closer
+	CloseStdin() (err error)
+	Pid() int
+	Stdio() (_ io.Writer, _ io.Reader, _ io.Reader)
+	ResizeConsole(width, height uint16) (err error)
+	Kill() (bool, error)
+	Signal(options interface{}) (bool, error)
+	Wait() (err error)
+	ExitCode() (_ int, err error)
+}
+
+// ProcessHost is the interface for creating processes.
+type ProcessHost interface {
+	CreateProcess(config interface{}) (Process, error)
+}
+
+// Container is the interface for container objects, either running on the host or
+// in a utility VM.
+type Container interface {
```

Comment:
Should we also have `ContainerHost` for the UVM set of interfaces that would include `CreateContainer`?

Reply:
So far I don't think we need it (there would be no users, and I think it's a Go anti-pattern to create unnecessary interfaces), but it might be interesting to create one later for the host itself to minimize the differences between Argon and Xenon.

Reply:
I guess since the plugin point is the bridge and not the HCS its safe to say that we can just maintain a fork for Argon/Xenon. Just seemed like having a `ContainerHost` for the Host OS and any UVM made it simpler but in practice it will be a duplicate impl of the interface anyways and we will have to special case because the other operations on a UVM Start/Stop/etc don't make sense for the Host OS.

Reply:
Right, hcsoci has to know how to add resources to UVMs, so we don't gain a lot by abstracting further.